lib/msal-browser/src/interaction_client/NativeInteractionClient.ts

```diff
@@ -42,14 +47,31 @@ export class NativeInteractionClient extends BaseInteractionClient {
 
         // start the perf measurement
         const nativeATMeasurement = this.performanceClient.startMeasurement(PerformanceEvents.NativeInteractionClientAcquireToken, request.correlationId);
+        const reqTimestamp = TimeUtils.nowSeconds();
+
+        // initialize native request
         const nativeRequest = await this.initializeNativeRequest(request);
 
+        // check if the tokens can be retrieved from internal cache
+        try {
+            const result = await this.acquireTokensFromCache(this.accountId, nativeRequest);
+            nativeATMeasurement.endMeasurement({
+                success: true,
+                isNativeBroker: true,
+                fromCache: true
+            });
+            return result;
+        } catch (e) {
+            // continue with a native call for any and all errors
+            this.logger.info("MSAL internal Cache does not contain tokens, proceed to make a native call");
```

Comment:
should you make a new error type for this?

Reply:
This is intentional as we anyway fall back to network call and there is nothing to handle. Did not want to add an error type when there is no next step of handling the specific error except convey the message(which logs do here)